shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -624,6 +624,7 @@ public static String getStreamForReaderGroup(String groupNameName) {
      */
     public static String validateUserStreamName(String name) {
         Preconditions.checkNotNull(name);
+        Preconditions.checkArgument(name.length()<256, "Name must have less than 256 characters");
```

Comment:
How does this work with multi-byte unicode character strings?

`String#length()` counts code units, not code points.

We may want to use `String#codePointCount(int, int)`. This will allow languages in higher unicode planes to also use 255 characters (which can consume up to 4 bytes per character in UTF8).

I believe the limiting factor in a scope name or stream name is the 1MB ZooKeeper ZNode size limit.

Will 255 character ever be too limiting? 255 characters is like 7 UUIDs.

Reply:
Some languages use multiple code points for a single character or grapheme. See: http://illegalargumentexception.blogspot.com/2009/05/java-rough-guide-to-character-encoding.html#javaencoding_strnlen

Reply:
We don't store Stream names in ZK. We stash them inside Table Segment Key-Value Pairs, either as Keys (max 8KB) or values (max 1MB - but may contain more than just a stream name). So the main constraint here will be Key - max 8KB.

Stream names are also part of the Segment names (`<Scope-Name>/<Stream-Name>/<Segment-Name>`. Given that we use Segment names as FileSystem paths (for NFS) or Object Keys (for S3/ExtS3), and so on, the constraint might be even higher than 8KB.

If you look at the line immediately following this line you'll see a Regex that enforces the name to be alphanumeric. As such, the character count is good enough because ASCII should be enough to encode this. I don't think we should worry about higher-level unicode chars in this case).

Reply:
In case of HDFS, the maximum component length for a file path is 255 bytes in UTF-8 encoding. 
[https://issues.apache.org/jira/browse/HDFS-6055](url)

Since these names are to only contain alphanumeric characters, I do not think using `String#codePointCount(int, int)` would offer significant benefits.

Reply:
If the maximum length for a path is 256, then your change isn't going to fix anything. See my previous comment why.

Reply:
@andreipaduroiu
> If the maximum length for a path is 256, then your change isn't going to fix anything. See my previous comment why.

I think this means each path part, not the entire path. He is sizing the max length to the limits of HDFS path parts.

@ldr7
> Since these names are to only contain alphanumeric characters, would using String#codePointCount(int, int) offer any benefits?

Yeah, my suggestion is invalid given the \p{Alnum} character class constraint. Apologies for the distraction! I can be a stickler about i18n, but I didn't notice it doesn't really apply in this context.

Now the only thing bugging me is we can't throw all constraint violations at once! ;)